contracts/JobRegistry.sol

```diff
@@ -178,6 +255,148 @@ contract JobRegistry {
         return _createJob(Status.Opened, senderId, senderId, 0, _jobDataUri);
     }
 
+    /**
+     * @notice Allows an employee to propose his service for a job
+     * @param _jobId The job linked to the new proposal
+     * @param _rateToken the token choose for the payment
+     * @param _rateAmount the amount of token choosed
+     * @param _proposalDataUri token Id to IPFS URI mapping
+     */
+    function createProposal(
+        uint256 _jobId,
+        address _rateToken,
+        uint256 _rateAmount,
+        string calldata _proposalDataUri
+    ) public {
+        uint256 senderId = tlId.walletOfOwner(msg.sender);
+        require(senderId > 0, "You sould have a TalentLayerId");
+
+        Job storage job = jobs[_jobId];
+        require(job.status == Status.Opened, "Job is not opened");
+        require(
+            job.proposals[senderId].employeeId != senderId,
+            "You already create a proposal for this job"
+        );
+        require(job.countProposals < 40, "Max proposals count reached");
+        require(
+            job.employerId != senderId,
+            "You couldn't create proposal for your own job"
+        );
+        require(
+            bytes(_proposalDataUri).length > 0,
+            "Should provide a valid IPFS URI"
+        );
+
+        job.countProposals++;
+        job.proposals[senderId] = Proposal({
+            status: ProposalStatus.Pending,
+            employeeId: senderId,
+            rateToken: _rateToken,
+            rateAmount: _rateAmount,
+            proposalDataUri: _proposalDataUri
+        });
+
+        emit ProposalCreated(
+            _jobId,
+            senderId,
+            _proposalDataUri,
+            ProposalStatus.Pending,
+            _rateToken,
+            _rateAmount
+        );
```

Comment:
`function getProposal()` takes `_proposalId` as parameter.
So we need to return `proposalId` here?

However, isn't it easier to get the proposal in`function getProposal()` with _jobId and _senderId?